File: code/functions/pyprism/pyprism/data/tests/test_load_bio.py
# -*- coding: utf-8 -*-
"""
@created: Dec 15 2020
@modified: Dec 15 2020
@author: Yoann Pradat

Tests for _load_bio.py module.
"""

import pandas as pd
from pyprism.data import load_bio

# User functions =======================================================================================================

def test_load_bio_met500():
    df_bio = load_bio(study="met500")
    assert type(df_bio) == pd.core.frame.DataFrame

    df_bio = load_bio(study="met500", mode="all")
    assert type(df_bio) == pd.core.frame.DataFrame

def test_load_bio_prism():
    df_bio = load_bio(study="prism")
    assert type(df_bio) == pd.core.frame.DataFrame

    df_bio = load_bio(study="prism", mode="all")
    assert type(df_bio) == pd.core.frame.DataFrame

def test_load_bio_tcga():
    df_bio = load_bio(study="tcga")
    assert type(df_bio) == pd.core.frame.DataFrame

    df_bio = load_bio(study="tcga", mode="all")
    assert type(df_bio) == pd.core.frame.DataFrame

    df_bio = load_bio(study="prism", identifiers=["201410666GH", "200905404EN"], identifiers_name="Subject_Id")
    assert type(df_bio) == pd.core.frame.DataFrame

    df_bio = load_bio(study="tcga", identifiers=["TCGA-02-0047", "TCCA-02-0055"], identifiers_name="Subject_Id")
    assert type(df_bio) == pd.core.frame.DataFrame

File: code/functions/rprism/tests/testthat/test-load_bio.R
# MET500

test_that("load_bio() met500 all works", {
  skip_on_cran()

  cat("\n")
  df_bio <- load_bio(study="met500")
  expect_true(is(df_bio, "data.frame"))

  cat("\n")
  df_bio <- load_bio(study="met500", mode="all")
  expect_true(is(df_bio, "data.frame"))
})

test_that("load_bio() met500 Subject_Id works", {
  skip_on_cran()

  cat("\n")
  df_bio <- load_bio(study="met500", identifiers=c("MO_1537", "MO_1196"), identifiers_name="Subject_Id")
  expect_true(is(df_bio, "data.frame"))
})



# PRISM

test_that("load_bio() prism all works", {
  skip_on_cran()

  cat("\n")
  df_bio <- load_bio(study="prism")
  expect_true(is(df_bio, "data.frame"))
})

test_that("load_bio() prism Subject_Id works", {
  skip_on_cran()

  cat("\n")
  df_bio <- load_bio(study="prism", identifiers=c("201410666GH", "200905404EN"), identifiers_name="Subject_Id")
  expect_true(is(df_bio, "data.frame"))
})


# TCGA

test_that("load_bio() tcga all works", {
  skip_on_cran()

  cat("\n")
  df_bio <- load_bio(study="tcga")
  expect_true(is(df_bio, "data.frame"))
})


test_that("load_bio() tcga Subject_Id works", {
  skip_on_cran()

  cat("\n")
  df_bio <- load_bio(study="tcga", identifiers=c("TCGA-02-0047", "TCCA-02-0055"), identifiers_name="Subject_Id")
  expect_true(is(df_bio, "data.frame"))
})
